integration/vm_factory/vm_cache_test.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 ARM Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+cidir=$(dirname "$0")
+source "${cidir}/../../metrics/lib/common.bash"
+
+kata_runtime=$(command -v kata-runtime)
+
+# Environment variables
+KATA_CACHE_INIT="${kata_runtime} factory init"
+KATA_CACHE_STATUS="${kata_runtime} factory status"
+KATA_CACHE_DESTROY="${kata_runtime} factory destroy"
+KATA_CACHE_SOCK="/var/run/kata-containers/cache.sock"
```

Comment:
It would be nice if we could use a random sock file instead of the default one. This way we will be testing if the factory init process reads it correctly from the `configuration.toml`  (would need to change the `enable_vm_cache_config()` properly).